Source/Scene/ModelExperimental/GeometryPipelineStage.js

```diff
@@ -334,14 +458,19 @@ function addAttributeDeclaration(shaderBuilder, attributeInfo) {
     glslType = attributeInfo.glslType;
   }
 
-  if (semantic === VertexAttributeSemantic.POSITION) {
+  const isPosition = semantic === VertexAttributeSemantic.POSITION;
+  if (isPosition) {
     shaderBuilder.setPositionAttribute(glslType, attributeName);
   } else {
     shaderBuilder.addAttribute(glslType, attributeName);
   }
+
+  if (isPosition && use2D) {
+    shaderBuilder.addAttribute("vec3", "a_position2D");
```

Comment:
I don't think this is an accurate name because "model coordinates" is supposed to be the positions with respect to the model. But the 2D coordinates are with respect to the world.

If you want I can name this `a_positionWC2D` ?

Reply:
Is it accurate to say they're in the model coordinate system? I know they're not in world space because they're with respect to the reference point, but how technical are we being here?

Reply:
I guess not, it's the 2D equiavelent of model space... okay, keep it as-is.